pymc/gp/util.py

```diff
@@ -88,29 +88,6 @@ def replace_with_values(vars_needed, replacements=None, model=None):
     return fn(**replacements)
 
 
-def infer_size(X, n_points=None):
-    R"""
-    Maybe attempt to infer the size, or N, of a Gaussian process input matrix.
-
-    If a specific shape cannot be inferred, for instance if X is symbolic, then an
-    error is raised.
-
-    Parameters
-    ----------
-    X: array-like
-        Gaussian process input matrix.
-    n_points: None or int
-        The number of rows of `X`.  If `None`, the number of rows of `X` is
-        calculated from `X` if possible.
-    """
-    if n_points is None:
-        try:
-            n_points = int(X.shape[0])
-        except TypeError:
-            raise TypeError("Cannot infer 'shape', provide as an argument")
-    return n_points
```

Comment:
If I recall, the purpose of this function was to catch errors where input `X` was a theano tensor or PyMC random variable, wherever shape was symbolic.  If so, the user would have to pass the shape in as an arg.  With the new Aesara, will `np.shape` work alright here, and make `infer_size` not necessary?

Reply:
`np.shape(tensor)` gives the `tensor.shape` symbolically and these days it's completely fine to have a symbolic `size` too

Reply:
wow, amazing.